### DISCUSSION TITLE
`eth_getBalance` intermittently returns `0x0` value

### DISCUSSION THREAD

**Discussion by SergeyVolynkin** [WalletConnect/walletconnect-monorepo#3868] [Developer Support]
Hi WalletConnect team,

We recently switched to `Sepolia testnet` (before we've been on Goerli) and started to observe flakiness in `eth_getBalance` periodically returning a falsy `0x0` value (where the actual balance is non-zero)

We've prepared code example to demo the issue:
https://github.com/SergeyVolynkin/wallet-connect-flaky-eth_getBalance-request

Recording of the issue:

https://github.com/WalletConnect/walletconnect-monorepo/assets/34238697/925939f4-30dd-4ac1-ab20-c7b6133a77b0

**Comment by SergeyVolynkin** [comment]
cc @glitch-txs for visibility 🙏

**Reply by glitch-txs** [reply]
Hi, I tested your example and it works fine on my end, I was able to connect and all balances fetched were correct, I would try to scope this issue a bit more, maybe try Wagmi alone without Web3Modal and see if the issue persists, you can go lower to wagmi/core or viem eventually, I would ask on their repo as well

**Reply by SergeyVolynkin** [reply]
Thank you for helping to investigate, @glitch-txs!

I am currently running the example repo and do not experience `0x0` flakiness anymore 

Is it possible that `POST https://rpc.walletconnect.com/v1/?chainId=eip155:11155111&projectId=1826…` requests were routed to some malfunctioning Sepolia RPC nodes? 🤔 When the `0x0` issue with `POST https://rpc.walletconnect.com/v1/…` reproduced consistently for us y-day, we compared with fetching via `Infura` endpoints –– which were stable at that time.

Is it possible to configure WalletConnect to route us to `Infura` nodes? (we can pay for our own Infura nodes, but Infura API KEY should be somehow stored securely in https://cloud.walletconnect.com/)

I see there is "secrets" section in WalletConnect Cloud, but no way to configure to use specific RPC provider(we'd go with Infura)
<img width="1160" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/34238697/a2cb3fb8-5dd8-4f94-9a40-fb3f68d79446">
